samples/sid_end_device/src/sidewalk.c

```diff
@@ -121,265 +108,40 @@ static void state_sidewalk_entry(void *o)
-			break;
-		}
-
-		memcpy(&sm->sid->last_status, p_status, sizeof(struct sid_status));
-		sid_hal_free(p_status);
-	} break;
-	case SID_EVENT_SEND_MSG: {
-		sidewalk_msg_t *p_msg = (sidewalk_msg_t *)sm->event.ctx;
-		if (!p_msg) {
-			LOG_ERR("sid send msg is NULL");
-			break;
-		}
-
-		e = sid_put_msg(sm->sid->handle, &p_msg->msg, &p_msg->desc);
-		if (e) {
-			LOG_ERR("sid send err %d", (int)e);
-		}
-		LOG_DBG("sid send (type: %d, id: %u)", (int)p_msg->desc.type, p_msg->desc.id);
-		int mutex_err = k_mutex_lock(&pending_message_list_mutex, K_FOREVER);
-		if (mutex_err != 0) {
-			LOG_ERR("Failed to lock mutex for message list");
-			break;
-		}
-		sys_slist_append(&pending_message_list, &p_msg->node);
-		k_mutex_unlock(&pending_message_list_mutex);
-	} break;
-	case SID_EVENT_CONNECT: {
-		if (!(sm->sid->config.link_mask & SID_LINK_TYPE_1)) {
-			LOG_ERR("Can not request connection - BLE not enabled");
-			return;
-		}
-		sid_error_t e = sid_ble_bcn_connection_request(sm->sid->handle, true);
-		if (e) {
-			LOG_ERR("sid conn req err %d", (int)e);
-		}
-	} break;
-	case SID_EVENT_FILE_TRANSFER: {
-#ifdef CONFIG_SIDEWALK_FILE_TRANSFER
-		struct data_received_args *args = (struct data_received_args *)sm->event.ctx;
-		if (!args) {
-			LOG_ERR("File transfer event data is NULL");
-			break;
-		}
-		LOG_INF("Received file Id %d; buffer size %d; file offset %d", args->desc.file_id,
-			args->buffer->size, args->desc.file_offset);
-		uint8_t hash_out[32];
-		sid_pal_hash_params_t params = { .algo = SID_PAL_HASH_SHA256,
-						 .data = args->buffer->data,
-						 .data_size = args->buffer->size,
-						 .digest = hash_out,
-						 .digest_size = sizeof(hash_out) };
-
-		sid_error_t e = sid_pal_crypto_hash(&params);
-		if (e != SID_ERROR_NONE) {
-			LOG_ERR("Failed to hash received file transfer with error %s",
-				SID_ERROR_T_STR(e));
-		} else {
-#define HEX_PRINTER(a, ...) "%02X"
-#define HEX_PRINTER_ARG(a, ...) hash_out[a]
-			char hex_str[sizeof(hash_out) * 2 + 1] = { 0 };
-			snprintf(hex_str, sizeof(hex_str), LISTIFY(32, HEX_PRINTER, ()),
-				 LISTIFY(32, HEX_PRINTER_ARG, (, )));
-			LOG_INF("SHA256: %s", hex_str);
-		}
-
-		sid_error_t ret = sid_bulk_data_transfer_release_buffer(
-			sm->sid->handle, args->desc.file_id, args->buffer);
-		if (ret != SID_ERROR_NONE) {
-			LOG_ERR("sid_bulk_data_transfer_release_buffer returned %s",
-				SID_ERROR_T_STR(ret));
-		}
-		sid_hal_free(args);
-#endif /* CONFIG_SIDEWALK_FILE_TRANSFER */
-	} break;
-	case SID_EVENT_LAST:
-		break;
-	}
-
-#ifdef CONFIG_SID_END_DEVICE_CLI
-	if (sm->event.id >= SID_EVENT_LAST) {
-		app_dut_event_process(sm->event, sm->sid);
+	if (!validate_event(sm, STATE_SIDEWALK)) {
+		return;
 	}
-#endif /* CONFIG_SID_END_DEVICE_CLI */
+	LOG_DBG("Handle event %s", sm->application_event.event_handler.name ?
+					   sm->application_event.event_handler.name :
+					   "UNKNOWN");
+	sm->application_event.event_handler.call[STATE_SIDEWALK](sm->application_event.ctx, sm);
```

Comment:
minor: consider simlification of this place. It is one of the most importatnt place

Reply:
`call[STATE]` is an actual event handler, it is inside the event object retrieved from the queue next to the context to pass to this event, and this event from queue is stored in the state machine object. 

I tried to do some aliases to simplify this call chain